packages/ec-core/src/operator/selector/mod.rs

```diff
@@ -289,26 +299,29 @@ where
 )]
 #[cfg(test)]
 mod tests {
-    use anyhow::Context;
-    use rand::{rngs::ThreadRng, thread_rng};
+    use std::convert::Infallible;
+
+    use rand::{rng, rngs::ThreadRng};
 
     use super::{Select, Selector};
     use crate::operator::{Composable, Operator};
 
     // A simple `Selector`` that always returns the first item in a vector.
     struct First;
 
+    #[derive(Debug, thiserror::Error)]
+    #[error("At least one individual in the population is required to make a selection from it")]
+    struct EmptyPopulationError;
```

Comment:
Is there a reason to create a new `EmptyPopulationError` in the various examples and tests in this file instead of using the `EmptyPopulation` error type that is used in the actual selectors?

Having `EmptyPopulationError` does illustrate how one might create an error type when making a new selector, but using the existing error type would highlight its existence and encourage people to use that instead of making a new one.

Thoughts?

Reply:
I'm inclined to use the existing error type instead of creating a new one.